packages/contracts/test/plugin/plugin-setup-processor.ts

```diff
@@ -467,82 +547,119 @@ describe('Plugin Setup Processor', function () {
-          );
+            proxy,
+            setupV1.address,
+            repoUUPS.address,
+            helpersV1,
+            EMPTY_DATA
+          )
+        ).to.be.revertedWith(customError('SetupAlreadyPrepared'));
+      });
 
-        await psp.applyInstallation(
+      it('returns the plugin, helpers, and permissions', async () => {
+        let returnedPluginAddress;
+        let returnedHelpers;
+        let uninstallPermissionsV1: PermissionOperation[];
+        ({
+          returnedPluginAddress: returnedPluginAddress,
+          returnedHelpers: returnedHelpers,
+          permissions: uninstallPermissionsV1,
+        } = await prepareUninstallation(
+          psp,
           targetDao.address,
-          pluginSetupBad,
-          pluginSetupMockRepoAddress,
-          plugin,
-          prepareInstallpermissions
-        );
+          proxy,
+          setupV1.address,
+          repoUUPS.address,
+          helpersV1,
+          EMPTY_DATA
+        ));
 
-        // prepare first uninstallation
-        const uninstallData = ethers.utils.defaultAbiCoder.encode(
-          ['bool'],
-          [false]
+        expect(returnedPluginAddress).to.equal(proxy);
+        expect(returnedHelpers).to.deep.equal(helpersV1);
+        expect(uninstallPermissionsV1).to.deep.equal(
+          mockPermissionsOperations(0, 1, Operation.Revoke).map(perm =>
+            Object.values(perm)
+          )
         );
+      });
 
-        await psp.prepareUninstallation(
-          targetDao.address,
-          plugin,
-          pluginSetupBad,
-          pluginSetupMockRepoAddress,
-          helpers,
-          uninstallData
-        );
+      it('prepares a UUPS upgradeable plugin uninstallation', async () => {
+        await expect(
+          prepareUninstallation(
+            psp,
+            targetDao.address,
+            proxy,
+            setupV1.address,
+            repoUUPS.address,
+            helpersV1,
+            EMPTY_DATA
+          )
+        ).to.not.be.reverted;
+      });
+
+      it('prepares a cloneable plugin uninstallation', async () => {
+        let clone;
+        let helpersC1;
+        let permissionsC1;
+
+        ({
+          plugin: clone,
+          helpers: helpersC1,
+          permissions: permissionsC1,
+        } = await installHelper(psp, targetDao, setupC1, repoClones));
 
-        // prepare second uninstallation
         await expect(
-          psp.prepareUninstallation(
+          prepareUninstallation(
+            psp,
             targetDao.address,
-            plugin,
-            pluginSetupBad,
-            pluginSetupMockRepoAddress,
-            helpers,
-            ethers.utils.defaultAbiCoder.encode(['bool'], [true])
+            clone,
+            setupC1.address,
+            repoClones.address,
+            helpersC1,
+            EMPTY_DATA
           )
-        ).to.be.revertedWith(customError('SetupAlreadyPrepared'));
+        ).to.not.be.reverted;
       });
     });
```

Comment:
we lack the test for `HelpersHashMismatch`.

Reply:
Added